### PR TITLE
Avoid classcastexception in correlation-id-middleware

### PR DESCRIPTION
**Pull request by shamsimam**

### CODE REVIEW

File: kitchen/src/kitchen/core.clj [hunk 1]
```diff
@@ -502,7 +502,11 @@
         (let [response (handler request)]
           (if (map? response)
             (add-cid-into-response request response)
-            (async/go (add-cid-into-response request (async/<! response)))))))))
+            (async/go
+              (let [nested-response (async/<! response)]
+                (if (map? nested-response)
```
Comment (dposada) [inline kitchen/src/kitchen/core.clj]: When would it not be a map?

Comment (shamsimam) [inline kitchen/src/kitchen/core.clj]: Websocket responses return channels that lead to classcastexceptions

Comment (dposada) [inline kitchen/src/kitchen/core.clj]: Should we add a comment explaining that?

Comment (shamsimam) [inline kitchen/src/kitchen/core.clj]: Okay, will add

Comment (shamsimam) [inline kitchen/src/kitchen/core.clj]: Done.

**Pull request merged**